lib/Platform/Intl/java/com/facebook/hermes/intl/PlatformDateTimeFormatterAndroid.java

```diff
@@ -204,8 +210,21 @@ else if (needTime)
   }
 
   @Override
-  public boolean isValidTimeZone(String timeZone) {
-    return TimeZone.getTimeZone(timeZone).getID().equals(timeZone);
+  public String normalizeValidTimeZone(final String timeZone) throws JSRangeErrorException {
+    Optional<String> normalizedValidTimeZone = allAvailableTimeZones.stream()
+      .filter(new Predicate<String>() {
+        @Override
+        public boolean test(String tz) {
+          return tz.equalsIgnoreCase(timeZone);
```

Comment:
I believe this is incorrect.  https://tc39.es/ecma402/#sec-case-sensitivity-and-case-mapping only allows mapping/folding ascii a-z <-> A-Z.  equalsIgnoreCase is probably applying additional locale-specific comparisons.

Reply:
You just need to bring this back: 

https://github.com/facebook/hermes/pull/571/files#diff-bd6622e75c91b32a4ac2f2a7f9ad2dc1e6c9ffa72c71e78070f3fc57c8d4f4e3L382-L397

Instead of using the built-in case insensitive comparison methods.